src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/AuthenticatorFactoryInterface.php

```diff
@@ -15,26 +15,29 @@
 use Symfony\Component\DependencyInjection\ContainerBuilder;
 
 /**
- * @method int getPriority() defines the position at which the authenticator is called
- *
  * @author Wouter de Jong <wouter@wouterj.nl>
  */
 interface AuthenticatorFactoryInterface
 {
     /**
-     * Creates the authenticator service(s) for the provided configuration.
-     *
-     * @return string|string[] The authenticator service ID(s) to be used by the firewall
+     * Defines the priority at which the authenticator is called.
      */
-    public function createAuthenticator(ContainerBuilder $container, string $firewallName, array $config, string $userProviderId);
+    public function getPriority(): int;
 
     /**
-     * Defines the configuration key used to reference the authenticator
+     * Defines the configuration key used to reference the provider
      * in the firewall configuration.
      *
      * @return string
      */
     public function getKey();
 
     public function addConfiguration(NodeDefinition $builder);
+
+    /**
+     * Creates the authenticator service(s) for the provided configuration.
+     *
+     * @return string|string[] The authenticator service ID(s) to be used by the firewall
+     */
+    public function createAuthenticator(ContainerBuilder $container, string $firewallName, array $config, string $userProviderId);
```

Comment:
Missing return type on purpose, to not break compat with 5.4?

Reply:
No, from a time where we didn't have union return types yet. I've added it (but honsetly I expect things to break for 5.4)